### PR TITLE
Add two additional format placeholders for beeversion

### PR DESCRIPTION
**Pull request by paulmenzel**

### CODE REVIEW

**Comment by mariux** [comment]
Isn't this possible by using %-e or %_e ? Can't test it right now because I am sitting at the beach ;)

**Comment by mariux** [comment]
Confirmed... please use %-e and %_e like it is used in the mentioned template.. 

git grep PKGEXTRAVERSION_UNDERSCORE

;)

**Comment by paulmenzel** [comment]
We evaluated that option too, but concluded that it wasn’t elegant for our use case.

To build the proprietary NVIDIA modules for the Linux kernel, currently the following code is used to normalize the version number, that means appending *.0* for Linux releases.

```
LINUXKLOCALVER=$(beeversion --format "%v.mx64.%r" "${LINUXPKG}")

# '4.4.mx64.75' to '4.4.0.mx64.75'
if [[ $LINUXKLOCALVER =~ ^([0-9]+\.[0-9]+)(\.mx64.*) ]]; then
    LINUXKLOCALVER="${BASH_REMATCH[1]}.0${BASH_REMATCH[2]}"
fi
```

Our goal is to have a generic script, also working with Linux release candidates, which have an extra version.

Without additional placeholders, we have to check if an extra version is set, and dependent on that add `-rcX`, which is not as elegant as the code with the placeholders.

```
# '4.4.mx64.75' to '4.4.0.mx64.75'
# '4.8-rc2.mx64.95' to '4.8.0-rc2.mx64.95'
if [[ $LINUXKLOCALVER =~ ^([0-9]+\.[0-9]+)(-?.*)(\.mx64.*) ]]; then
    LINUXKLOCALVER="${BASH_REMATCH[1]}.0${BASH_REMATCH[2]}${BASH_REMATCH[3]}"
fi
```

**Comment by mariux** [comment]
how does the code difference look like? actually your code is '%d' == '-%e' which is infact != '%-e' (if '%e' == '') so what is '%d' good for anyway in the given code examples (it's not even used in the example (neither is %e used in the example given))? 

your code *always* adds a dash in front of %e... you get this by prefixing a dash in the formatstring where %e is used ('-%e')... don't you? so what is the commit helping you to do? please give complete beeversion examples with expected and current output..

using the example from the commit: 
```
beeversion --format="%v%d" linux-4.8_rc2-95.x86_64
4.8-rc2

beeversion --format="%v-%e" linux-4.8_rc2-95.x86_64
4.8-rc2

beeversion --format="%v%-e" linux-4.8_rc2-95.x86_64
4.8-rc2
```
where the first two examples are identical...and the third only adds the dash if %e is set:
```
beeversion --format="%v%d\n" linux-4.8-95.x86_64
4.8-

beeversion --format="%v-%e\n" linux-4.8-95.x86_64
4.8-

beeversion --format="%v%-e\n" linux-4.8-95.x86_64
4.8
```

sorry but this is nothing that will help you to have a generic script for whatever problem you try to solve... it just always adds a dash... that you could prefix in the format string anyway... or what do i miss here?

i think you get an even more generic version by using the proposed '%-e' solution because it works with and without an extraversion while the '%d' only works when extraversion is set.

works for me ;)

File: src/bee_version_output.c [hunk 1]
```diff
@@ -105,6 +105,12 @@ void print_format(char* s, struct beeversion *v, char *filter_pkgfullname)
                 case 'e':
                     printf("%s", v->extraversion);
                     break;
+                case 'E':
+                    printf("_%s", v->extraversion);
```
Comment (mariux) [inline src/bee_version_output.c]: this is equal to '_%e'

File: src/bee_version_output.c [hunk 2]
```diff
@@ -105,6 +105,12 @@ void print_format(char* s, struct beeversion *v, char *filter_pkgfullname)
...
+                    break;
+                case 'd':
+                    printf("-%s", v->extraversion);
```
Comment (mariux) [inline src/bee_version_output.c]: this is equal to `-%e`

**Comment by paulmenzel** [comment]
You are right. I missed that you put the dash or underscore after the *%*. Useful feature. Unfortunately, not something I would have expected.

Thank you for making me aware of it.

Seeing the two environment variables, I thought to have a placeholder for each of them.

```
> beeversion linux-4.8_rc2-95.x86_64
PKGNAME=linux
PKGEXTRANAME=
PKGEXTRANAME_UNDERSCORE=
PKGEXTRANAME_DASH=
PKGVERSION=( 4.8 4 4.8 )
PKGEXTRAVERSION=rc2
PKGEXTRAVERSION_UNDERSCORE=_rc2
PKGEXTRAVERSION_DASH=-rc2
PKGREVISION=95
PKGARCH=x86_64
PKGFULLNAME=linux
PKGFULLVERSION=4.8_rc2
PKGFULLPKG=linux-4.8_rc2-95
PKGALLPKG=linux-4.8_rc2-95.x86_64
PKGSUFFIX=
```

**Comment by paulmenzel** [comment]
I updated my branch with the forgotten if statements, so that it now works as claimed.

**Comment by mariux** [comment]
but still unneeded addition since `%-e` and `%_e` work fine and is now fool proof (#208).

**Pull request closed without merging**